database-commons/src/main/java/io/cdap/plugin/db/CommonSchemaReader.java

```diff
@@ -90,6 +90,10 @@ public Schema getSchema(ResultSetMetaData metadata, int index) throws SQLExcepti
       case Types.DECIMAL:
         int precision = metadata.getPrecision(index); // total number of digits
         int scale = metadata.getScale(index); // digits after the decimal point
+        if (scale == 0) {
+          throw new SQLException(new UnsupportedTypeException(String.format("Unsupported SQL Type: %s with scale 0.",
```

Comment:
Why are we wrapping this with sqlexception. It is thrown in case there is error accessing database
https://docs.oracle.com/javase/7/docs/api/java/sql/SQLException.html#:~:text=An%20exception%20that%20provides%20information,available%20via%20the%20method%20getMesasge%20.

Reply:
This is what method currently throws - we based it on the following line https://github.com/data-integrations/database-plugins/blob/540d5a56f9d33888fee27bc6f5846132fff8dd7a/database-commons/src/main/java/io/cdap/plugin/db/CommonSchemaReader.java#L130
Is there a better solution? Also if there is - how to handle it on other callers as this method is being used from all the other db plugins?

Reply:
There is no need to wrap `UnsupportedTypeException ` into `SQLException`. we can just have unsupported type exception

Reply:
changing it to throw UnsupportedTypeException also updates base method signature which is used by all database plugins

Reply:
I see. Looks like we are handling it like that in this method.